packages/flutter/lib/src/rendering/flex.dart

```diff
@@ -299,10 +320,14 @@ class RenderFlex extends RenderBox with ContainerRenderObjectMixin<RenderBox, Fl
     int totalFlex = 0;
     int totalChildren = 0;
     assert(constraints != null);
-    final double mainSize = (_direction == FlexDirection.horizontal) ? constraints.constrainWidth() : constraints.constrainHeight();
-    final bool canFlex = mainSize < double.INFINITY && mainAxisAlignment != MainAxisAlignment.collapse;
-    double crossSize = 0.0;  // This is determined as we lay out the children
-    double freeSpace = canFlex ? mainSize : 0.0;
+    final bool isHorizontal = _direction == FlexDirection.horizontal;
+    final double minMainSize = isHorizontal ? constraints.constrainWidth(0.0) : constraints.constrainHeight(0.0);
+    final double maxMainSize = isHorizontal ? constraints.constrainWidth() : constraints.constrainHeight();
+    final double mainSize = mainAxisSpace == MainAxisSpace.max ? maxMainSize : minMainSize;
```

Comment:
Maybe call this availableSize ?  It won't always end up being the size in the main axis.